src/middlewared/middlewared/plugins/nfs_/krb5.py

```diff
@@ -0,0 +1,95 @@
+import errno
+from middlewared.service import private, Service
+from middlewared.service_exception import CallError
+from middlewared.plugins.idmap import DSType
+from middlewared.plugins.directoryservices import DSStatus
+from middlewared.schema import accepts, Dict, Str
+
+
+class NFSService(Service):
+
+    class Config:
+        service = "nfs"
+        service_verb = "restart"
+        datastore_prefix = "nfs_srv_"
+        datastore_extend = 'nfs.nfs_extend'
+
+    @private
+    async def validate_directoryservices(self, ds):
+        """
+        Validation currently only succeeds in case of HEALTHY AD
+        directory service. At future point, depending on demand, the
+        same convenience feature can be added for the LDAP directory
+        service.
+        """
+        ds = await self.middleware.call('directoryservices.get_state')
+        ad_status = DSStatus[ds['activedirectory']]
+        ldap_status = DSStatus[ds['ldap']]
+
+        if ldap_status != DSStatus.DISABLED:
+            raise CallError('Automated addition of NFS SPN records has '
```

Comment:
Is this a correct error message?

Reply:
It's unclear and pre-coffee english. In that regard, no... it's barely grammatical. I'll simplify the message.